datastore/key.go

```diff
@@ -254,6 +255,20 @@ func DecodeKey(encoded string) (*Key, error) {
 
 	ref := new(pb.Reference)
 	if err := proto.Unmarshal(b, ref); err != nil {
+		// If there was an err on the key lets try to decode the new key type by default check to see if key converter
+		// has been implemented.
+		if keyConversionProject == "" {
+			nKey, nLibKeyErr := keycompat.DecodeToNewKey(encoded)
+			if nLibKeyErr != nil {
+				// returning the orginal error on purpose
+				return nil, err
+			}
+			oKey, err := convertNewKeyFormatToOldKeyFormat(nKey)
```

Comment:
return convertNewKey...(...)

no need for the if statement

Reply:
fixed